### PR TITLE
[SP-51] Download Layered Assets

### PR DESCRIPTION
**Pull request by chuwong35122**
- Download canvas as Zip of PNGs
- Create a modal that allow user to config Canvas Size (the size is based on the uploaded assets)

### CODE REVIEW

File: components/generate/layer/LayerAssetPreviewCard.tsx [hunk 1]
```diff
@@ -66,58 +70,72 @@ function LayerAssetPreviewCard({ asset, layerIndex, assetIndex }: Props) {
...
+              />
+            </Box>
+            <Box sx={{ borderBottom: 2 }} />
+
+            <Box
+              sx={{ padding: 2, backgroundColor: grey[300], height: "auto" }}
+            >
               <motion.div variants={textVariant}>
                 <Typography
                   component="p"
```
Comment (pointmekin) [inline components/generate/layer/LayerAssetPreviewCard.tsx:104]: `<Typogrpahy />` uses paragraph element as a default. No need to declare menually.

File: components/generate/layer/LayerSettingModal.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    setHeights(_heights)
+  }, [layers])
+
+  function handleChangeWidth(w: number | string) {
+    const _dimension = { ...assetDimension }
+    if (typeof w === "number") {
+      _dimension.width = w
+    } else if (typeof w === "string") {
+      _dimension.width = parseInt(w)
+    }
```
Comment (pointmekin) [inline components/generate/layer/LayerSettingModal.tsx]: _dimension.width = parseInt(`${w}`)

File: components/generate/layer/LayerSettingModal.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    setAssetDimension(_dimension)
+  }
+
+  function handleChangeHeight(h: number | string) {
+    const _dimension = { ...assetDimension }
+    if (typeof h === "number") {
+      _dimension.height = h
+    } else if (typeof h === "string") {
+      _dimension.width = parseInt(h)
+    }
```
Comment (pointmekin) [inline components/generate/layer/LayerSettingModal.tsx]: same here

File: components/generate/layer/LayerSettingModal.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+  return (
+    <Modal
+      open={isOpen}
+      onClose={() => setIsOpen(false)}
+      aria-labelledby="setting-modal"
+      aria-describedby="setting-modal"
+    >
+      <Box sx={style}>
+        <div>{JSON.stringify(assetDimension)}</div>
```
Comment (pointmekin) [inline components/generate/layer/LayerSettingModal.tsx:76]: Forgot to remove?

File: components/generate/layer/LayerSettingModal.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                <MenuItem key={i} value={width}>{`${width} px`}</MenuItem>
+              ))}
+            </Select>
+          </Stack>
+          <Stack
+            direction="row"
+            justifyContent="space-between"
+            alignItems="center"
+          >
+            <Typography component="p" sx={{ fontWeight: 500 }}>
```
Comment (pointmekin) [inline components/generate/layer/LayerSettingModal.tsx:109]: no need to say component="p" here as well, but it's alright

File: components/generate/layer/steps/CreateLayeredTicketStep6.tsx [hunk 6]
```diff
@@ -75,23 +93,25 @@ function CreateLayeredTicketStep6() {
...
-                draggable={false}
-              />
-            </motion.div>
-          )}
-          containerStyles={{ backgroundColor: indigo[300] }}
-        />
+        <div onClick={handleDownloadAssetFiles}>
+          <ImageLabelCard
+            title="Download Generated Assets"
+            description="Tame these tickets onto your computer storage, unleash it when you are ready to."
```
Comment (pointmekin) [inline components/generate/layer/steps/CreateLayeredTicketStep6.tsx:99]: "Tame"?

Comment (chuwong35122) [reply components/generate/layer/steps/CreateLayeredTicketStep6.tsx:99]: YES... TAME
Will re-write these wordings later

**Review by pointmekin — APPROVED** [review]
GOGO

**Pull request merged**